src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBaseTest.java

```diff
@@ -277,4 +287,59 @@ void testTableStream(MtAmazonDynamoDbBase mtDynamoDb) {
         }
     }
 
+    /**
+     * Verifies that reading virtual tables with streams disabled doesn't return records.
+     */
+    @ParameterizedTest
+    @ArgumentsSource(Args.class)
+    void testDisabledStreams(MtAmazonDynamoDbBase mtDynamoDb) {
+        final MtAmazonDynamoDbStreams mtDynamoDbStreams = MtAmazonDynamoDbStreams.createFromDynamo(mtDynamoDb,
+            AmazonDynamoDbLocal.getAmazonDynamoDbStreamsLocal());
+        try {
+            String tenant = TENANTS[0];
+
+            // create table with streams enabled
+            String tableWithStreamsEnabled = TENANT_TABLE_NAME + "_streams_enabled";
+            CreateTableRequest createTableRequestStreamsEnabled =
+                newCreateTableRequest(tableWithStreamsEnabled)
+                    .withStreamSpecification(new StreamSpecification()
+                        .withStreamEnabled(true)
+                        .withStreamViewType(NEW_AND_OLD_IMAGES));
+            MT_CONTEXT.withContext(tenant, () -> mtDynamoDb.createTable(createTableRequestStreamsEnabled));
+
+            // create table with streams disabled
+            String tableWithStreamsDisabled = TENANT_TABLE_NAME + "_streams_disabled";
+            CreateTableRequest createTableRequestStreamsDisabled =
+                newCreateTableRequest(tableWithStreamsDisabled)
+                    .withStreamSpecification(new StreamSpecification()
+                        .withStreamEnabled(false)
+                    );
+            MT_CONTEXT.withContext(tenant, () -> mtDynamoDb.createTable(createTableRequestStreamsDisabled));
+
+            // put an item in each table
+            final MtRecord streamsEnabledRecord = putTestItem(mtDynamoDb, tableWithStreamsEnabled, tenant, 1);
+            putTestItem(mtDynamoDb, tableWithStreamsDisabled, tenant, 2);
+
+            // verify that the record from the stream enabled table is returned
+            MT_CONTEXT.withContext(tenant, () -> {
+                String streamArn = mtDynamoDb.describeTable(tableWithStreamsEnabled).getTable().getLatestStreamArn();
+                assertNotNull(streamArn);
+                String iterator = getShardIterator(mtDynamoDbStreams, streamArn).get();
+                assertGetRecords(mtDynamoDbStreams, iterator, streamsEnabledRecord);
+            });
+
+            // verify that either the streamArn is null(for ByTable or ByAccount) or the iterator returns no records
+            MT_CONTEXT.withContext(tenant, () -> {
+                String streamArn = mtDynamoDb.describeTable(tableWithStreamsDisabled).getTable().getLatestStreamArn();
```

Comment:
Should the streamArn returned here be null if queried in the context of a tenant for which the arn is disabled?

Reply:
It seems like with a native AmazonDynamoDB instance, if you describe a table with streaming disabled, then getLatestStreamArn() returns null.  So if we want this method to behave consistently with native behavior(and ByTable), then yes, it should always return null at the line you're referring to.